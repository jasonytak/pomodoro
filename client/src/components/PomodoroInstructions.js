import React from 'react';

const PomodoroInstructions = () => {
  return (
    <div className="pomodoro-instructions">
        <ol>
          <ol>1. Decide on a task.</ol>
          <ol>2. Start the Pomodoro Timer.</ol>
          <ol>3. Work on the task for 25 minutes.</ol>
          <ol>4. When timer rings, pause work and take a 5 minute break.</ol>
          <ol>5. Repeat!</ol>
        </ol>
    </div>
  );
};

export default PomodoroInstructions;
