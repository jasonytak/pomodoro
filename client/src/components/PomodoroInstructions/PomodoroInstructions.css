.pomodoro-instructions {
  margin-top: 5%;
  text-align: center;
  margin-right: 5%;
}

.pomodoro-instructions li {
  color: #eebe97;
  font-family: 'Source Sans Pro', sans-serif;
  font-size: calc(15px + 0.5vw);
  list-style-type: none;
}