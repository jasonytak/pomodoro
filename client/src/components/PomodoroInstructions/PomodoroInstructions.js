import './PomodoroInstructions.css';
import React from 'react';

const PomodoroInstructions = ({ totalWorkMinutes, totalBreakMinutes }) => {
  return (
    <div className="pomodoro-instructions">
      <ol>
        <ol>1. DECIDE ON A TASK</ol>
        <ol>2. START THE POMODORO TIMER</ol>
        <ol>3. WORK ON THE TASK FOR {totalWorkMinutes} MINUTES</ol>
        <ol>4. WHEN TIMER RINGS, PAUSE WORK AND TAKE A {totalBreakMinutes} MINUTE BREAK</ol>
        <ol>5. REPEAT!</ol>
      </ol>
    </div>
  );
};

export default PomodoroInstructions;
