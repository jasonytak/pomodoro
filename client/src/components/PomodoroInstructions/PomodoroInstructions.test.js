import React from 'react';
import { shallow } from 'enzyme';
import PomodoroInstructions from './PomodoroInstructions';

let wrapped;

beforeEach(() => {
  wrapped = shallow(<PomodoroInstructions />);
});

it('renders totalWorkMinutes prop properly', () => {
  wrapped.setProps({ totalWorkMinutes: 20 });
  expect(
    wrapped.containsMatchingElement(<p>3. WORK ON THE TASK FOR 20 MINUTES</p>)
  ).toBeTruthy();
});

it('renders totalBreakMinutes prop properly', () => {
  wrapped.setProps({ totalBreakMinutes: 10 });
  expect(
    wrapped.containsMatchingElement(
      <p>4. WHEN TIMER RINGS, PAUSE WORK AND TAKE A 10 MINUTE BREAK</p>
    )
  ).toBeTruthy();
});
