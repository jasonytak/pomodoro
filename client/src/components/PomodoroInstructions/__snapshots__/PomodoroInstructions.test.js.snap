// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly rendered text component 1`] = `
<div
  className="pomodoro-instructions"
>
  <p>
    1. DECIDE ON A TASK
  </p>
  <p>
    2. START THE POMODORO TIMER
  </p>
  <p>
    3. WORK ON THE TASK FOR 
     MINUTES
  </p>
  <p>
    4. WHEN TIMER RINGS, PAUSE WORK AND TAKE A 
     MINUTE BREAK
  </p>
  <p>
    5. REPEAT!
  </p>
</div>
`;
